pallets/identity/src/lib.rs

```diff
@@ -127,7 +127,28 @@ pub mod pallet {
 
             ensure!(Self::is_identity_owned_by_sender(&who, &identity_id), Error::<T>::IdentityNotOwned);
 
-            <IdentityTraitList<T>>::insert(identity_id, key, value);
+            <IdentityTraitList<T>>::try_mutate(identity_id, key, |v| -> DispatchResult {
+                *v = value;
+                Ok(())
+            })?;
+
+            Self::deposit_event(Event::IdentityUpdated(identity_id, who));
+
+            Ok(())
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        /// Edit an existing identity trait
+        pub fn edit_identity_trait(origin: OriginFor<T>, identity_id: u32, key: Vec<u8>, value: Vec<u8>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            ensure!(Self::is_identity_owned_by_sender(&who, &identity_id), Error::<T>::IdentityNotOwned);
+
+            <IdentityTraitList<T>>::try_mutate(identity_id, key, |v| -> DispatchResult {
```

Comment:
Are these two functions identical?

Reply:
yes -- that is why I was suggesting we make it one function and call it `upsert`

Reply:
`try_mutate` makes it possible to handle both insert and update using the same code.

Reply:
Yep that makes more sense now. I knew the `try_mutate` section would be the same but I didn't realize fully the extent to which the functions were legitimately just the same thing. I do think `upsert` is a bit of an unclear name - maybe `modify`?

Reply:
`modify` delivers the concept of editing, or updating -- but I don't feel like it delivers the idea of "add" or "insert".

I am not aware of another term typically used to denote the action of both an "update" and an "insert" other than "upsert"

Reply:
we could also mirror the internal method we are using, that is `mutate_identity_trait`

Reply:
Yeah, fair point well made. Modify isn't really sufficient, is it. I usually just use "insert_or_update", "add_or_update", my main hesitation is I've never actually seen "upsert" used that I can remember.

Reply:
> we could also mirror the internal method we are using, that is `mutate_identity_trait`

Yeah mutate had occurred to me too, and I actually had the same concern as you brought up with `modify`.